### DISCUSSION TITLE
jtablesaw cant read sqlite database correctly

### DISCUSSION THREAD

**Discussion by dddeveloperrr** [jtablesaw/tablesaw#1056] [Q&A]
hi
I have a sqlite database which has a table with two columns.  Name (which is Text type)  and  ID (Which is Integer type).

![Untitled](https://user-images.githubusercontent.com/11577640/159127217-e951c6ac-d806-443a-a724-3de7d2058567.jpg)

![image](https://user-images.githubusercontent.com/11577640/159126937-019b8731-52bf-4aac-962b-7e2a463615ad.png)

I want to read this database and print it's content to console. here is my code :

```
package Main;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import tech.tablesaw.api.Table;

public class Main {

	public static void main(String[] args) {

		try {

			String DB_URL = "jdbc:sqlite:/home/farhad/Desktop/test";
			Connection conn = DriverManager.getConnection(DB_URL);
			Statement stmt = conn.createStatement();
			String query = "SELECT * FROM mytable";
			ResultSet results = stmt.executeQuery(query);
			Table tb = Table.read().db(results, "mytable");

			System.out.println(tb.printAll());

		} catch (Exception e) {
			System.out.println(e.getLocalizedMessage());
		}

	}
}
```

and here is output:
```

         mytable         
 Name   |      ID       |
-------------------------
  Jack  |    734294471  |
  Alex  |   1468588942  |
 Ashly  |  -2092083883  |

```

as you can see, it returns wrong output

**Comment by lwhite1** [comment]
That looks like an integer overload problem.  The values are too large for
a Java int.  They should map to long.  It’s possible SQLite doesn’t provide
enough detail on the size to map correctly.   Could you retry with values
under 2 billion and let me know what you get?

**Comment by dddeveloperrr** [comment]
it works fine with small numbers

![image](https://user-images.githubusercontent.com/11577640/159134795-d5151885-80ad-4830-a18e-7865a3877037.png)


and here is the output:
```

        mytable         
 Name   |      ID      |
------------------------
  Jack  |      123456  |
  Alex  |         123  |
 Ashly  |  1999999999  |

```
